equinox/nn/attention.py

```diff
@@ -0,0 +1,136 @@
+from typing import Optional
+
+import jax
+import jax.numpy as jnp
+import jax.random as jrandom
+import numpy as np
+
+from ..custom_types import Array
+from ..module import Module, static_field
+from .dropout import Dropout
+from .linear import Linear
+
+
+class MultiheadAttention(Module):
+    """
+    Multihead Attention layer from 'Attention Is All You Need' (https://arxiv.org/abs/1706.03762)
+    """
+
+    embed_dim: int = static_field()
+    num_heads: int = static_field()
+    kdim: int = static_field()
+    vdim: int = static_field()
+    _qkv_same_embed_dim: bool = static_field()
+    head_dim: int = static_field()
+    q_proj: Linear
+    k_proj: Linear
+    v_proj: Linear
+    out_proj: Linear
+    dropout: Dropout
+
+    def __init__(
+        self,
+        embed_dim: int,
+        num_heads: int,
+        dropout: float = 0.0,
+        use_bias: bool = True,
+        kdim: Optional[int] = None,
+        vdim: Optional[int] = None,
+        add_bias_kv: bool = False,
+        *,
+        key: "jax.random.PRNGKey",
```

Comment:
The convention so far has been to add `**kwargs` and pass them on through `super().__init__(**kwargs)`, in case of co-operative multiple inheritance.